lang/typescript/src/utils/regions.test.ts

```diff
@@ -32,3 +55,207 @@ describe('region yaml loader', () => {
     expect(config).toBeNull();
   });
 });
+
+describe('getConcatenationRules', () => {
+  describe('with address object', () => {
+    test('returns undefined if combined_address_format is not defined', () => {
+      const config = getRegionConfig('US');
+      const address: Address = {
+        countryCode: 'US',
+        address2: 'Apt 1',
+      };
+      expect(config).not.toBeNull();
+      expect(
+        getConcatenationRules(config!, address, 'address2'),
+      ).toBeUndefined();
+    });
+
+    test('returns default rules if no language specific rules are defined', () => {
+      const config = getRegionConfig('BR');
+      const address: Address = {
+        countryCode: 'BR',
+        line2: 'dpto 4',
+        neighborhood: 'Centro',
+      };
+      expect(config).not.toBeNull();
+      expect(getConcatenationRules(config!, address, 'address2')).toEqual([
+        {key: 'line2'},
+        {key: 'neighborhood', decorator: ', '},
+      ]);
+    });
+
+    test('returns default rules if script specific rules are defined but script is not detected', () => {
+      const config = getRegionConfig('TW');
+      const address: Address = {
+        countryCode: 'TW',
+        line2: '30號',
+        neighborhood: '大甲區',
+      };
+      expect(config).not.toBeNull();
+      expect(getConcatenationRules(config!, address, 'address2')).toEqual([
+        {key: 'line2'},
+        {key: 'neighborhood'},
+      ]);
+    });
+
+    test('returns default rules if script specific rules but multiple scripts detected', () => {
```

Comment:
love the test coverage

Reply:
Whoop typo in this too, should be `specifies`